python/packages/ai/teams/ai/ai.py

```diff
@@ -32,14 +42,11 @@ class AI(Generic[StateT]):
     _options: AIOptions
     _log: Logger
     _actions: Dict[str, ActionEntry[StateT]] = {}
-    _review_prompt: Optional[
-        Callable[[TurnContext, StateT, PromptTemplate], Awaitable[Optional[Plan]]]
-    ] = None
-    _review_plan: Optional[Callable[[TurnContext, StateT, Plan], Awaitable[Plan]]] = None
 
-    def __init__(self, options: AIOptions, log=Logger("teams.ai")) -> None:
+    def __init__(self, options: AIOptions, *, log=Logger("teams.ai")) -> None:
         self._options = options
         self._log = log
+        self._options.planner.log = log
```

Comment:
Maybe check whether the planner has a logger first to avoid overriding user defined logger?
By the way, should we just rely on developer to set the logger? If developers do not set a logger, it means they don't want logging. Current code seems will always add log for developers.

Reply:
I want the developer to be able to pass the logger into the `App` and have that logger passed down to all sub components for logging similar to how `slack` does it. This way if they want they can implement their own logger. I don't think the developer should have to pass the logger instance into every class below the app explicitly.